### PR TITLE
feat(adapters): add Cline adapter

### PR DESCRIPTION
**Pull request by Alex-ai-future**
## Summary

Add support for [Cline](https://github.com/cline/cline) (formerly Claude Dev), a VS Code AI coding extension.

## Changes

- **New adapter**: `src/adapters/cline.rs` - Implements `SourceAdapter` trait for Cline
- **Registration**: `src/adapters/mod.rs` - Registers the new adapter

## Data Source

- **Path**: `~/Library/Application Support/Code/User/globalStorage/saoudrizwan.claude-dev/tasks/`
- **Format**: Each task is a directory containing `ui_messages.json` (conversation data)
- **Directory naming**: Timestamp-based (milliseconds)

## Features

| Capability | Status |
|------------|--------|
| Auto-discovery | ✅ |
| Full index | ✅ |
| Incremental sync | ✅ |
| FTS5 keyword search | ✅ |
| Semantic search | ✅ |
| Source filter | ✅ |
| Time range filter | ✅ |
| Resume in original CLI | — (VS Code extension, no CLI) |

## Testing

9 unit tests covering:
- Message parsing (user/assistant role mapping)
- Incremental sync (skip unchanged, reparse on mtime change, discover new)
- Edge cases (empty messages, invalid JSON, non-timestamp directories)

## Usage

```bash
recall sync    # Sync all sessions including Cline
recall         # Launch TUI - Cline sessions appear with `CL` label
recall search Q # Search across all sessions
```

### CODE REVIEW

**Comment by Alex-ai-future** [comment]
@samzong ready

File: src/adapters/cline.rs [hunk 1]
```diff
@@ -0,0 +1,592 @@
...
+                            });
+                        }
+                    }
+                    _ => {}
+                }
+            }
+            "ask" => {
+                if let Some(text) = extract_text(&msg) {
+                    let timestamp = msg.get("ts").and_then(|v| v.as_i64());
+                    result.push(RawMessage { role: Role::User, content: text, timestamp });
```
Comment (samzong) [inline src/adapters/cline.rs]: It doesn't seem appropriate here to set it to User.
suggestion use `Role::Assistant`

Comment (Alex-ai-future) [reply src/adapters/cline.rs]: fixed

**Comment by samzong** [comment]
@Alex-ai-future Thanks for u do amazing things!
